tests/qeqiskit/optimizer/optimizer_test.py

```diff
@@ -46,3 +62,49 @@ def test_optimizer_succeeds_on_cost_function_without_gradient(
         assert "opt_value" in results
         assert "opt_params" in results
         assert "history" in results
+
+
+class TestQiskitSinusoidalOptimizerTests(OptimizerTests):
+    """Some optimizers assume that the cost function can be expressed as a sum of
+    products of sines and cosines (Eq. 12 of arXiv:1903.12166), a functional form which
+    reflects the cost function of many variational quantum algorithms. Such optimizers
+    may fail to find the minimum of cost functions that are not of this form.
+
+    This class removes the tests from the base class that check whether the optimizer
+    finds the minimum of such non-sinusoidal cost functions, and replaces them with a
+    test on a sinuisoidal cost function."""
+
+    @pytest.fixture
+    def sum_cos_x(self):
+        """Create a sinuisoidal function whose minimum is at the origin with a value of
+        zero."""
+        return lambda x: -np.sum(np.cos(x)) + x.shape[0]
+
+    def test_optimizer_succeeds_with_optimizing_rosenbrock_function(self):
+        pass
```

Comment:
I think `pytest.xfail("message") would be more appropriate?

Reply:
Added `pytest.skip` in 4db10fa8bca5c63c975e7682fe7364c70dba01f2.